envoy/http/codec.h

```diff
@@ -324,6 +324,25 @@ class StreamCallbacks {
   virtual void onBelowWriteBufferLowWatermark() PURE;
 };
 
+/**
+ * An adapter for an HTTP Stream that is tightly coupled with both the
+ * lifetime of the stream and low-level events that can occur on the stream.
+ */
+class StreamAdapter {
```

Comment:
Naming nit. It is unclear why this is an 'adapter'. I think it can just be `CodecEvents` or `StreamEvents`. If you decide to rename it, please also rename the setter method.